Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -91,114 +162,79 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
 
         blur.snp.makeConstraints { make in
-            make.edges.equalTo(self.view)
+            make.edges.equalTo(view)
         }
 
         suggestionCell.delegate = self
 
         NotificationCenter.default.addObserver(self, selector: #selector(dynamicFontChanged), name: .dynamicFontChanged, object: nil)
     }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        reloadSearchEngines()
+        reloadData()
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        // The height of the suggestions row may change, so call reloadData() to recalculate cell heights.
+        coordinator.animate(alongsideTransition: { _ in
+            self.tableView.reloadData()
+        }, completion: nil)
+    }
+
+    private func animateSearchEnginesWithKeyboard(_ keyboardState: KeyboardState) {
+        layoutSearchEngineScrollView()
+
+        UIView.animate(withDuration: keyboardState.animationDuration, animations: {
+            UIView.setAnimationCurve(keyboardState.animationCurve)
+            self.view.layoutIfNeeded()
+        })
+    }
 
     @objc func dynamicFontChanged(_ notification: Notification) {
         if notification.name == .dynamicFontChanged {
             reloadData()
         }
     }
-
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        reloadSearchEngines()
-        reloadData()
-    }
     
+    // MARK: Internal
+
     private func setupSearchEngineScrollViewIfNeeded() {
         if !hasQuickSearchEngines { return }
 
-        searchEngineScrollView.layer.shadowRadius = 0
-        searchEngineScrollView.layer.shadowOpacity = 100
-        searchEngineScrollView.layer.shadowOffset = CGSize(width: 0, height: -SearchViewControllerUX.searchEngineTopBorderWidth)
-        searchEngineScrollView.layer.shadowColor = SearchViewControllerUX.searchEngineScrollViewBorderColor
-        searchEngineScrollView.clipsToBounds = false
-
-        searchEngineScrollView.decelerationRate = UIScrollView.DecelerationRate.fast
         view.addSubview(searchEngineScrollView)
-
-        searchEngineScrollViewContent.layer.backgroundColor = UIColor.clear.cgColor
         searchEngineScrollView.addSubview(searchEngineScrollViewContent)
 
         layoutTable()
         layoutSearchEngineScrollView()
 
         searchEngineScrollViewContent.snp.makeConstraints { make in
-            make.center.equalTo(self.searchEngineScrollView).priority(10)
+            make.center.equalTo(searchEngineScrollView).priority(10)
             // Left-align the engines on iphones, center on ipad
             if UIScreen.main.traitCollection.horizontalSizeClass == .compact {
-                make.left.equalTo(self.searchEngineScrollView).priority(1000)
+                make.left.equalTo(searchEngineScrollView).priority(1000)
```

Comment:
Should not be using 1000. Should be using `UILayoutPriorityRequired` unless there is some reason for it not working?

Reply:
Let me check and be sure about it, layout is not the best layout there honestly

Reply:
Priorities Changed from numeric values to use levels